src/actions/mod.rs

```diff
@@ -35,6 +35,21 @@ use std::time::Duration;
 
 use self::compiler_message_parsing::{FileDiagnostic, ParseError, Suggestion};
 
+// TODO: Support non-`file` URI schemes in VFS. We're currently ignoring them because
+// we don't want to crash the RLS in case a client opens a file under different URI scheme
+// like with git:/ or perforce:/ (Probably even http:/? We currently don't support remote schemes).
+macro_rules! ignore_non_file_uri {
```

Comment:
> Maybe call this macro inside the span parsing code and propagate an error later?

I already propagate the error [here](https://github.com/rust-lang-nursery/rls/pull/430/files#diff-53ac0aa2d451315631bac6f85bcfb7bdR770). Do you mean something different?

> If that is too much work we could have a macro that parses a uri string to a path or does the error handling like this and another than just does this.

Do you mean something like separate `parse_file_path_ignore!` and `convert_pos_to_span_ignore!` kind of macros that do what the original functions do (without the `unwrap`) but still expand to include `return;` at call site if a resulting error is `UrlFileParseError::InvalidScheme`?

I think it's okay as a short-term fix, because it's somewhat flexible and doesn't compilate existing parsing/conversion functions, but I'm fine with whatever and can change it, just tell me which change would you prefer :slightly_smiling_face: